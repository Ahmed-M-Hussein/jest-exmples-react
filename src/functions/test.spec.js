import React from "react";
import { create , act} from "react-test-renderer";
import DemoComponent from "../Demos/DemoComponent";
import { iteratee } from "lodash";
describe("My Testing Scope", () => {
  //   it("add 2 numbers 1, 4", () => {
  //     expect(add(1, 4)).toBe(5);
  //   });
  //   it("test add function without params", () => {
  //     expect(add()).toBe(NaN);
  //   });
  //   it("test add with negative number -2 4", () => {
  //     expect(add(-2,4)).toBe(2);
  //   });
  //   it("test add with string parms 1 4", () => {
  //     expect(add("1","4")).toBe(5);
  //   });
  //   it("test add with object parms ", () => {
  //     expect(add({},"4")).toBe(NaN);
  //   });
  //   it("test add with object parms ", () => {
  //     expect(add({},{})).toBe(NaN);
  //   });
  //   it("test add with 2 params 1 ,4  ", (done) => {
  //     function callback(result) {
  //       try {
  //         expect(result).toBe(5);
  //         done();
  //       } catch (error) {
  //         done(error);
  //       }
  //     }
  //     add(1, 4, callback);
  //   });

  //   it("test add with object parms ", () => {
  //     return add(1, 4).then((result) => {
  //       expect(result).toBe(5);
  //     });
  //   });

  //   it("test add with object parms ", async () => {

  //    const result = await add(1, 4)
  //    expect(result).toBe(5)
  //   });

  it("Test DemoComponent empty props", () => {
    const wrapper = create(<DemoComponent />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("Test DemoComponent empty props", () => {
    const wrapper = create(<DemoComponent />);
    const instance = wrapper.getInstance();

    instance.handlePressBtn();
    instance.handlePressBtn();
    instance.handlePressBtn();

    expect(instance.state.counter).toBe(3);
  });


  it("Test DemoComponent empty props", () => {
    
    let wrapper 

    act(()=>{

        wrapper= create(<DemoComponent />);
    })
   

    act(()=>{

        wrapper.update(<DemoComponent name="Ahmed" />)
    })

    act(()=>{

        wrapper.update(<DemoComponent name="Mohamed" />)
    })

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

});



// function add  (x,y){

//     return x+y
// }



// z=1
// function addx  (x,y){

//     return x+y+z
// }
