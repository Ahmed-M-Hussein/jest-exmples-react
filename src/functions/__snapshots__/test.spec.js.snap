// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Testing Scope Test DemoComponent empty props 1`] = `
<View>
  <Text>
     Welcom 
    Guest
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`My Testing Scope Test DemoComponent empty props 2`] = `
<View>
  <Text>
     Welcom 
    Ahmed
     
  </Text>
  <Text>
     You Clicked this Btn 
    0
     times 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
